Client/Frontend/Browser/Playlist/Cache/PlaylistDownloadManager.swift

```diff
@@ -364,7 +414,9 @@ private class PlaylistFileDownloadManager: NSObject, URLSessionDownloadDelegate
         activeDownloadTasks[task] = MediaDownloadTask(id: item.pageSrc, name: item.name, asset: asset)
         task.resume()
         
-        delegate?.onDownloadStateChanged(streamDownloader: self, id: item.pageSrc, state: .inProgress, displayName: asset.displayNames(for: asset.preferredMediaSelection), error: nil)
+        DispatchQueue.main.async {
```

Comment:
For sake of synchronization can you also use `ensureMainThread` here. Also there couple of them above ^

Reply:
I don't want `ensureMainThread` because I need it `Dispatch` at a later time. Calling `ensureMainThread` does NOT guarantee an `async` dispatch. It only guarantees that it is either ran on main thread synchronously OR async depending on the calling context.

This is why I have it as `DispatchQueue.main.async`